pyanaconda/modules/security/installation.py

```diff
@@ -208,13 +208,30 @@ def run(self):
             return
 
         try:
-            selinux_cfg = SimpleConfigFile(os.path.join(self._sysroot, self.SELINUX_CONFIG_PATH))
-            selinux_cfg.read()
-            selinux_cfg.set(("SELINUX", self.SELINUX_STATES[self._selinux_mode]))
-            selinux_cfg.write()
+            # Read the SELinux configuration file.
+            path = os.path.join(self._sysroot, self.SELINUX_CONFIG_PATH)
+
+            with open(path, "r") as f:
+                lines = f.readlines()
+
+            # Modify the SELinux configuration.
+            lines = list(map(self._process_line, lines))
+
+            # Write the modified configuration.
+            with open(path, "w") as f:
+                f.writelines(lines)
+
         except OSError as msg:
             log.error("SELinux configuration failed: %s", msg)
 
+    def _process_line(self, line):
+        """Process a line from the SELinux configuration file."""
+        if line.startswith("SELINUX="):
```

Comment:
I think we should do the `trim` first. It could have for any reason and the config will be still valid but we won't match.

Also I'm thinking if we don't want to have this logged to know that we changed something at all?

Reply:
Fixed.